lib/e_chronicler_web/controllers/journal_entry_controller.ex

```diff
@@ -66,4 +66,13 @@ defmodule EChroniclerWeb.JournalEntryController do
     end
   end
 
+  def delete(conn, %{"id" => id}) do
+    journal_entry = JournalEntry.get_journal_entry(id)
+    {:ok, _journal_entry} = JournalEntry.delete_journal_entry(journal_entry)
```

Comment:
What about the error case when the given entry id does not exist?

Reply:
Also, I think it makes more sense for your `delete_journal_entry` function to take the id as a parameter and handle getting the entry inside itself.

Reply:
I'm having trouble capturing the error case in a test, I'm getting this when I put in a bad ID in the test:
`   ** (FunctionClauseError) no function clause matching in Ecto.Repo.Schema.delete/4`

But I'm unable to capture that in my current test, which is this: 
```
  test "delete_journal_entry/1 with invalid id returns error" do
    assert {:error, FunctionClauseError} = JournalEntry.delete_journal_entry(1000)
  end
```

I'm sure part of that is because the error is thrown by Ecto, not by the `delete_journal_entry` function itself - so is the move to add cases for the interior `Repo.get(JournalEntry, id)` call?

Reply:
Yes, you need to provide matches for both the `{:ok, ...}` and `{:error, ...}` cases from `Repo.get()`.

Reply:
I was looking into the [docs ](https://hexdocs.pm/ecto/Ecto.Repo.html#c:get/3)and it seems like `Repo.get()` doesn't return the `{:ok, ...}/{:error, ...}` cases, but rather either the resource or `nil`. But I'm having trouble passing the resource that it returned in the case that it is returned to Repo.delete() - getting a `no local return error.` Have been debugging for a while but am feeling stuck, but will try again after lunch.